includes/Checker/Checks/Enqueued_Styles_Scope_Check.php

```diff
@@ -0,0 +1,234 @@
+	 *               need any constructor arguments, it would just be an empty array.
+	 */
+	public function get_shared_preparations() {
+		$demo_posts = array_map(
+			static function( $post_type ) {
+				return array(
+					'post_title'   => "Demo {$post_type} post",
+					'post_content' => 'Test content',
+					'post_type'    => $post_type,
+					'post_status'  => 'publish',
+				);
+			},
+			$this->get_viewable_post_types()
+		);
+
+		return array(
+			Demo_Posts_Creation_Preparation::class => array( $demo_posts ),
+		);
+	}
+
+	/**
+	 * Runs the check on the plugin and amends results.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check results to amend and the plugin context.
+	 */
+	public function run( Check_Result $result ) {
+		$this->run_for_urls(
+			$this->get_urls(),
+			function ( $url ) use ( $result ) {
+				$this->check_url( $result, $url );
+			}
+		);
+
+		if ( ! empty( $this->plugin_styles ) ) {
+			$url_count = count( $this->get_urls() );
+			foreach ( $this->plugin_styles as $plugin_style ) {
+				if ( isset( $plugin_style['count'] ) && ( $url_count === $plugin_style['count'] ) ) {
+					$result->add_message(
+						false,
+						__( 'This style is being loaded in all contexts.', 'plugin-check' ),
+						array(
+							'code' => 'EnqueuedStylesScope.StyleLoadedInAllContext',
+							'file' => $plugin_style['path'],
+						)
+					);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Gets the list of URLs to run this check for.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of URL strings (either full URLs or paths).
+	 *
+	 * @throws Exception Thrown when a post type URL cannot be retrieved.
+	 */
+	protected function get_urls() {
+		$urls = array( home_url( '/' ) );
+
+		foreach ( $this->get_viewable_post_types() as $post_type ) {
+			$posts = get_posts(
+				array(
+					'posts_per_page' => 1,
+					'post_type'      => $post_type,
+					'post_status'    => array( 'publish', 'inherit' ),
+				)
+			);
+
+			if ( ! isset( $posts[0] ) ) {
+				throw new Exception(
+					sprintf(
+						/* translators: %s: The Post Type name. */
+						__( 'Unable to retrieve post URL for post type: %s', 'plugin-check' ),
+						$post_type
+					)
+				);
+			}
+
+			$urls[] = get_permalink( $posts[0] );
+
+			// Add post category link.
+			if ( 'post' === get_post_type( $posts[0] ) ) {
+				$category = get_the_category( $posts[0]->ID );
+				$urls[]   = get_category_link( $category[0]->term_id );
+			}
+		}
+
+		return $urls;
+	}
+
+	/**
+	 * Amends the given result by running the check for the given URL.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 * @param string       $url    URL to run the check for.
+	 *
+	 * @throws Exception Thrown when the check fails with a critical error (unrelated to any errors detected as part of
+	 *                   the check).
+	 */
+	protected function check_url( Check_Result $result, $url ) {
```

Comment:
What does the $url parameter do? I can't see it used.

Reply:
Nice catch. remove it in https://github.com/10up/plugin-check/pull/188/commits/112f11490c3b550b69d0b0e146979fae20c27c52